data/lua/core/interface.lua

```diff
@@ -60,6 +60,10 @@ if wesnoth.kernel_type() == "Game Lua Kernel" then
 	wesnoth.end_turn = wesnoth.deprecate_api('wesnoth.end_turn', 'wesnoth.interface.end_turn', 1, nil, wesnoth.interface.end_turn)
 	wesnoth.get_viewing_side = wesnoth.deprecate_api('wesnoth.get_viewing_side', 'wesnoth.interface.get_viewing_side', 1, nil, wesnoth.interface.get_viewing_side)
 	wesnoth.message = wesnoth.deprecate_api('wesnoth.message', 'wesnoth.interface.add_chat_message', 1, nil, wesnoth.interface.add_chat_message)
+	-- wesnoth.wml_actions.print doesn't exist yet at this point, so create a helper function instead.
+	wesnoth.print = wesnoth.deprecate_api('wesnoth.print', 'wesnoth.interface.add_overlay_text', 1, nil, function(cfg)
```

Comment:
IIRC, the plan was for no new Lua deprecations in 1.18.

Reply:
I'd hesitate to call it a deprecation if it's just a helper wrapper for something that doesn't exist right now anyway (assuming I'm understanding the comment right.

Reply:
The comment means that this code runs before the `print` function is defined.

We can certainly void the deprecation and hold off to 1.20 to deprecate it, but that creates the weird situation of having two functions in two different places that do essentially the same thing, so I don't think that's a great idea.